homeassistant/components/ezviz/siren.py

```diff
@@ -0,0 +1,133 @@
+    ]
+
+    async_add_entities(
+        EzvizSirenEntity(coordinator, camera, SIREN_ENTITY_TYPE)
+        for camera in coordinator.data
+        for capability, value in coordinator.data[camera]["supportExt"].items()
+        if capability == str(SupportExt.SupportActiveDefense.value)
+        if value != "0"
+    )
+
+
+class EzvizSirenEntity(EzvizBaseEntity, SirenEntity, RestoreEntity):
+    """Representation of a EZVIZ Siren entity."""
+
+    _attr_supported_features = SirenEntityFeature.TURN_ON | SirenEntityFeature.TURN_OFF
+    _attr_should_poll = False
+    _attr_assumed_state = True
+
+    def __init__(
+        self,
+        coordinator: EzvizDataUpdateCoordinator,
+        serial: str,
+        description: SirenEntityDescription,
+    ) -> None:
+        """Initialize the Siren."""
+        super().__init__(coordinator, serial)
+        self._attr_unique_id = f"{serial}_{description.key}"
+        self.entity_description = description
+        self._attr_is_on = False
+        self._delay_listener: Callable | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        if not (last_state := await self.async_get_last_state()):
+            return
+        self._attr_is_on = last_state.state == STATE_ON
+
+        if self._attr_is_on:
+            evt.async_call_later(self.hass, OFF_DELAY, self.off_delay_listener)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off camera siren."""
+        try:
+            result = await self.hass.async_add_executor_job(
+                self.coordinator.ezviz_client.sound_alarm, self._serial, 1
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Failed to turn siren off for {self.name}"
+            ) from err
+
+        if result:
+            if self._delay_listener is not None:
+                self._delay_listener()
+                self._delay_listener = None
+
+            self._attr_is_on = False
+            self.async_write_ha_state()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on camera siren."""
+        try:
+            result = self.hass.async_add_executor_job(
+                self.coordinator.ezviz_client.sound_alarm, self._serial, 2
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Failed to turn siren on for {self.name}"
+            ) from err
+
+        if result:
+            if self._delay_listener is not None:
+                self._delay_listener()
+                self._delay_listener = None
+
+            self._attr_is_on = True
+            self._delay_listener = evt.async_call_later(
+                self.hass, OFF_DELAY, self.off_delay_listener
+            )
+            self.async_write_ha_state()
+
+    @callback
+    def off_delay_listener(self, now: datetime) -> None:
```

Comment:
We don't allow calculating state in Home Assistant. The device should update its state. Please move the update to the 3rd party library.